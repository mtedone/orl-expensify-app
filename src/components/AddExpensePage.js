import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

// Refactors dispatch(addExpense) here so the component is more testable
const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense)),
});

// First argument is mapStateToProps, the second is mapDispatchToProps
export default connect(undefined, mapDispatchToProps)(AddExpensePage);
