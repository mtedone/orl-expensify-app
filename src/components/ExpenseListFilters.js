import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount } from '../actions/filters';

const ExpenseListFilters = (props) => (
  <div>
    <input
      type="text"
      value={props.filters.text}
      onChange={(e) => {
        // dispatch is available from components
        props.dispatch(setTextFilter(e.target.value));
      }}
    />
    <select
      value={props.filters.sortBy}
      onChange={(e) => {
        const val = e.target.value;
        if (val === 'date') {
          props.dispatch(sortByDate());
        } else if (val === 'amount') {
          props.dispatch(sortByAmount());
        }
      }}
    >
      <option value="date">Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
);

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
