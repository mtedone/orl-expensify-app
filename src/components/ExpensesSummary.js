import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
  const expenseWord = props.expensesCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(props.expensesTotal / 100).format(
    '$0,0.00'
  );
  return (
    <div>
      <h1>
        Viewing {props.expensesCount} {expenseWord} totalling{' '}
        {formattedExpensesTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(state.expenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
