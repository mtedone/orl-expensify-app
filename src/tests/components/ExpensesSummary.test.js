import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import selectExpensesTotal from '../../selectors/expenses-total';

test('should display ExpensesSummary correctly with one expense', () => {
  const total = selectExpensesTotal([expenses[0]]);
  const wrapper = shallow(
    <ExpensesSummary expensesCount={1} expensesTotal={total} />
  );
  expect(wrapper.debug()).toMatchSnapshot();
});

test('should display ExpensesSummary correctly with many expenses', () => {
  const total = selectExpensesTotal([expenses[0], expenses[1]]);
  const wrapper = shallow(
    <ExpensesSummary expensesCount={2} expensesTotal={total} />
  );
  expect(wrapper.debug()).toMatchSnapshot();
});
