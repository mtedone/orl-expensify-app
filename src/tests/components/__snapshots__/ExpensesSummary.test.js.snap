// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display ExpensesSummary correctly with many expenses 1`] = `
"<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling
     
    $1,097.00
  </h1>
</div>"
`;

exports[`should display ExpensesSummary correctly with one expense 1`] = `
"<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling
     
    $2.00
  </h1>
</div>"
`;
